### PR TITLE
add support for batch writes

### PR DESCRIPTION
**Pull request by DaMatrix**
this is really only api-level support. the region implementations remain unchanged, as none of them would benefit noticeably from this.

### CODE REVIEW

File: src/main/java/cubicchunks/regionlib/MultiUnsupportedDataException.java [hunk 1]
```diff
@@ -21,17 +21,20 @@
...
-	private final Throwable t;
+public class MultiUnsupportedDataException extends IOException {
+    private final Map<?, UnsupportedDataException> children;
 
-	public WrappedException(Throwable t) {
-		this.t = t;
-	}
+    public MultiUnsupportedDataException(Map<?, UnsupportedDataException> children) {
+        this.children = children;
+    }
```
Comment (Niko-sk2x) [inline src/main/java/cubicchunks/regionlib/MultiUnsupportedDataException.java:35]: Use `addSuppressed` instead of adding children yourself

**Review by Niko-sk2x — CHANGES_REQUESTED** [review]

File: src/main/java/cubicchunks/regionlib/MultiUnsupportedDataException.java (hunk 1 above)
Comment (DaMatrix) [reply src/main/java/cubicchunks/regionlib/MultiUnsupportedDataException.java:35]: but `SaveSectionException` does the same thing... i guess i should make it print them, though.

Comment (Niko-sk2x) [reply src/main/java/cubicchunks/regionlib/MultiUnsupportedDataException.java:35]: Yes, I didn't know about suppressed exceptions when I wrote it originally.

Comment (DaMatrix) [reply src/main/java/cubicchunks/regionlib/MultiUnsupportedDataException.java:35]: the thing is, i'm not suppressing these exceptions - i actually need them to be in a `Map`.

Comment (Niko-sk2x) [reply src/main/java/cubicchunks/regionlib/MultiUnsupportedDataException.java:35]: You could wrap each of them and specify the key it's for in a message. And from javadoc of addSuppressed:
```
Appends the specified exception to the exceptions that were suppressed in order to deliver this exception. This method is thread-safe and typically called (automatically and implicitly) by the try-with-resources statement.
(...)
Note that programmer written code is also able to take advantage of calling this method in situations where there are multiple sibling exceptions and only one can be propagated.
```
You are suppressing them, to later throw a different exception.

JVM uses it in try-with-resources when closing the resource while handling an exception throws.

Comment (DaMatrix) [reply src/main/java/cubicchunks/regionlib/MultiUnsupportedDataException.java:35]: i guess i could add an optional `key` parameter to `UnsupportedDataException`, but that seems even more hacky imo

edit: didn't see your message until after i sent this

**Pull request merged**